server/service/queries.go

```diff
@@ -287,17 +291,32 @@ func (svc *Service) ModifyQuery(ctx context.Context, id uint, p fleet.QueryPaylo
 	}
 	if p.Logging != nil {
 		query.Logging = *p.Logging
+		if *p.Logging != fleet.LoggingSnapshot {
```

Comment:
I believe we should check if: it's changing AND it's changing to snapshot:
```go
if query.Logging != *p.Logging && *p.Logging != fleet.LoggingSnapshot {
  shouldDiscardQueryResults = true
}
```

Reply:
Gotcha; this again was assuming that because these are pointers, we're only setting them when we have an update. I can change this though!